deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java

```diff
@@ -127,23 +135,46 @@ void registerFonts(BuildProducer<NativeImageResourcePatternsBuildItem> nativeIma
 
     @BuildStep(onlyIf = IsDevelopment.class)
     ReportRootBuildItem defaultReportRoot() {
+
         return new ReportRootBuildItem(DEFAULT_ROOT_PATH);
     }
 
     @BuildStep(onlyIf = IsDevelopment.class)
     void watchReportFiles(BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
+            BuildProducer<ReportFileBuildItem> reportFiles,
             List<ReportRootBuildItem> reportRoots) {
-        watchedPaths.produce(HotDeploymentWatchedFileBuildItem.builder().setLocationPredicate(path -> {
-            for (ReportRootBuildItem root : reportRoots) {
+
+        for (ReportRootBuildItem reportRoot : reportRoots) {
+            Path startDir = Paths.get(reportRoot.getPath()); // Specify your starting directory
```

Comment:
Maybe check if the path is inside the project directory? Not sure if this may become a security issue